Automattic/jetpack
# Retry changelog filename with suffixes if exists

If you run `tools/check-intra-monorepo-deps.sh -u`, and it has a filename collision when it auto-generates the changelog entry, then the script bails. This can leave the repo in a bad state, e.g. by failing to commit some auto-generated changelogs even if you retry.

This change attempts up to five numeric suffixes of the changelog filename before bailing.

#### Changes proposed in this Pull Request:
* Retry dependency-update changelog filenames with suffixes so that script is more robust

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Fork this branch
* Generate a changelog entry using `jetpack changelogger add ${packagename}` for a package with an outdated internal dependency. Use the default filename (which is the branch name)
* Run `tools/check-intra-monorepo-deps.sh -u`
* It should not have a filename conflict, but rather should automatically commit the changelog entry to `${branchname}-1`
